zio-http/src/main/scala/zio/http/Flash.scala

```diff
@@ -0,0 +1,392 @@
+    }
+
+  private def getMessageHtml[A: Schema, B: Schema](f: A => Html, g: B => Html): Flash[Message[Html, Html]] =
+    getMessage[A, B].map {
+      case Message.Notice(a)                                 => Message.Notice(f(a))
+      case Message.Alert(b)                                  => Message.Alert(g(b))
+      case Message.Both(Message.Notice(a), Message.Alert(b)) => Message.Both(Message.Notice(f(a)), Message.Alert(g(b)))
+    }
+
+  /**
+   * Creates a `Flash.Message` using the default keys for notice and alert.
+   *
+   * Additionally the values must be of type `String` so they can be transformed
+   * to `Html`.
+   *
+   * Usage e.g.: `Flash.getMessageHtml.foldHtml(showNotice, showAlert)(_ ++ _)`
+   */
+  def getMessageHtml: Flash[Message[Html, Html]] =
+    getMessageHtml[String, String](a => Dom.text(a), b => Dom.text(b))
+
+  /**
+   * Gets any flash value of type `A` with the given key `key`.
+   */
+  def get[A: Schema](key: String): Flash[A] = Flash.Get(Schema[A], key)
+
+  /**
+   * Gets a flash value of type `String` with the given key `key`.
+   */
+  def getString(key: String): Flash[String] = get[String](key)
+
+  /**
+   * Gets a flash value of type `A` associated with the notice key.
+   */
+  def getNotice[A: Schema]: Flash[A] = get[A](Message.Notice.name)
+
+  /**
+   * Gets a flash value of type `A` associated with the alert key.
+   */
+  def getAlert[A: Schema]: Flash[A] = get[A](Message.Alert.name)
+
+  /**
+   * Gets a flash value of type `Float` with the given key `key`.
+   */
+  def getFloat(key: String): Flash[Float] = get[Float](key)
+
+  /**
+   * Gets a flash value of type `Double` with the given key `key`.
+   */
+  def getDouble(key: String): Flash[Double] = get[Double](key)
+
+  /**
+   * Gets a flash value of type `Int` with the given key `key`.
+   */
+  def getInt(key: String): Flash[Int] = get[Int](key)
+
+  /**
+   * Gets a flash value of type `Long` with the given key `key`.
+   */
+  def getLong(key: String): Flash[Long] = get[Long](key)
+
+  /**
+   * Gets a flash value of type `UUID` with the given key `key`.
+   */
+  def getUUID(key: String): Flash[UUID] = get[UUID](key)
+
+  /**
+   * Gets a flash value of type `Boolean` with the given key `key`.
+   */
+  def getBoolean(key: String): Flash[Boolean] = get[Boolean](key)
+
+  /**
+   * Gets the first flash value of type `A` regardless of any key.
+   */
+  def get[A: Schema]: Flash[A] = withInput { map =>
+    map.keys.map(a => Flash.get(a)(Schema[A])).reduce(_ <> _)
+  }
+
+  private[http] def run[A](flash: Flash[A], sourceRequest: Request): Either[Throwable, A] =
+    sourceRequest
+      .cookie(COOKIE_NAME)
+      .toRight(new RuntimeException("flash cookie doesn't exist"))
+      .flatMap { cookie =>
+        try {
+          val content =
+            URLDecoder.decode(cookie.content, java.nio.charset.Charset.forName("UTF-8").toString.toLowerCase)
```

Comment:
Can't you just use `StandardCharsets.UTF_8`? 🤔
```suggestion
            URLDecoder.decode(cookie.content, StandardCharsets.UTF_8)
```

Reply:
Thanks, somehow I did not know about `StandardCharsets`